### PR TITLE
Usage queue poc

### PR DESCRIPTION
**Pull request by shimonewman**
The usage queue is constructed in two pieces:
 A worker and a swoole ticker that aggregates stats and fires on a fixed interval.
The worker sync behavior versus the ticker async behavior creates a conflict when the queue is busy and generates.
Swoole errors, which result in corrupted data.
In an attempt to solve it, we  refactored the architecture to use two queues :
The first queue aggregates the metrics based on the number of keys || last sent timestamp (the one that happens first)
and send them to the second queue, which inserts the stats into the DB.

The new architecture uses two sync elements and eliminates the necessary  to use a static array  that also conflicts
with multithreading.

From the test (e2e , manuel, k6) the results look promising

### CODE REVIEW

File: src/Appwrite/Platform/Workers/UsageDump.php [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                                '$id' => $id,
+                                'period' => $period,
+                                'time' => $time,
+                                'metric' => $key,
+                                'value' => $value,
+                                'region' => App::getEnv('_APP_REGION', 'default'),
+                            ]));
+                        } catch (Duplicate $th) {
+                            if ($value < 0) {
+                                //Todo debug (to be removed later @shimon)
```
Comment (eldadfux) [inline src/Appwrite/Platform/Workers/UsageDump.php:89]: Remove left overs

**Review by eldadfux — CHANGES_REQUESTED** [review]

File: src/Appwrite/Platform/Workers/Usage.php [hunk 2]
```diff
@@ -4,22 +4,22 @@
...
-        'inf' => '0000-00-00 00:00'
-    ];
-
-    protected const INFINITY_PERIOD = '_inf_';
+    private array $stats = [];
+    private int $lastTriggeredTime = 0;
+    private int $keys = 0;
+    private const INFINITY_PERIOD = '_inf_';
+    private const KEYS_THRESHOLD = 5;
+    private const KEYS_SENT_THRESHOLD = 60;
```
Comment (christyjacob4) [inline src/Appwrite/Platform/Workers/Usage.php:19]: Lets increase these values

File: src/Appwrite/Platform/Workers/Usage.php [hunk 3]
```diff
@@ -4,22 +4,22 @@
 
 use Exception;
 use Utopia\CLI\Console;
-use Utopia\Database\Database;
-use Utopia\Database\DateTime;
 use Utopia\Database\Document;
 use Utopia\Platform\Action;
+use Appwrite\Event\UsageDump;
 use Utopia\Queue\Message;
```
Comment (christyjacob4) [inline src/Appwrite/Platform/Workers/Usage.php:11]: Add a todo that we still need to handle the case of deployments performed when usage us being aggregated.

Comment (christyjacob4) [reply src/Appwrite/Platform/Workers/Usage.php:11]: Add another Todo to refactor the names of the two usage workers

File: src/Appwrite/Platform/Workers/Usage.php (hunk 2 above)
Comment (christyjacob4) [inline src/Appwrite/Platform/Workers/Usage.php:19]: Lets use this  variable we already have
 $interval = (int) App::getEnv('_APP_USAGE_AGGREGATION_INTERVAL', '60000');

File: app/controllers/api/users.php [hunk 4]
```diff
@@ -1189,15 +1189,16 @@ function createUser(string $hash, mixed $hashOptions, string $userId, ?string $e
         // clone user object to send to workers
         $clone = clone $user;
 
-        $dbForProject->deleteDocument('users', $userId);
-
-        $queueForDeletes
-            ->setType(DELETE_TYPE_DOCUMENT)
-            ->setDocument($clone);
+        $affected = $dbForProject->deleteDocument('users', $userId);
```
Comment (christyjacob4) [inline app/controllers/api/users.php:1192]: Lets move this change to a separate PR, where we can apply this change to all the endpoints that require it

File: docker-compose.yml [hunk 5]
```diff
@@ -84,6 +84,7 @@ services:
       - ./public:/usr/src/code/public
       - ./src:/usr/src/code/src
       - ./dev:/usr/src/code/dev
+      #- ./vendor/utopia-php/database:/usr/src/code/vendor/utopia-php/database
```
Comment (christyjacob4) [inline docker-compose.yml:87]: lets remove this line

**Review by christyjacob4 — APPROVED** [review]

**Pull request merged**